src/platform/forms/save-in-progress/actions.js

```diff
@@ -265,49 +265,40 @@ export function fetchInProgressForm(
   // TODO: Migrations currently aren’t sent; they’re taken from `form` in the
   //  redux store, but form.migrations doesn’t exist (nor should it, really)
   return (dispatch, getState) => {
-    const trackingPrefix = getState().form.trackingPrefix;
+    const { trackingPrefix } = getState().form;
     const apiUrl = inProgressApi(formId);
 
     // Update UI while we’re waiting for the API
     dispatch(setFetchFormPending(prefill));
 
     // Query the api and return a promise (for navigation / error handling afterward)
-    return fetch(apiUrl, {
-      credentials: 'include',
-      headers: {
-        'Content-Type': 'application/json',
-        'X-Key-Inflection': 'camel',
-        'Source-App-Name': window.appName,
+    return apiRequest(
+      apiUrl,
+      {
+        credentials: 'include',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Key-Inflection': 'camel',
+          'Source-App-Name': window.appName,
+        },
       },
-    })
-      .then(res => {
-        if (res.ok) {
-          return res.json();
-        }
-
-        // Make me a switch?
-        let status = LOAD_STATUSES.failure;
-        if (res.status === 401) {
-          dispatch(logOut());
-          status = LOAD_STATUSES.noAuth;
-        } else if (res.status === 403) {
-          status = LOAD_STATUSES.forbidden;
-        } else if (res.status === 404) {
-          status = LOAD_STATUSES.notFound;
-        }
-        return Promise.reject(status);
-      })
+      null,
+      null,
+      true,
+    )
       .then(resBody => {
         // Just in case something funny happens where the json returned isn’t an object as expected
         // Unfortunately, JavaScript is quite fiddly here, so there has to be additional checks
         if (typeof resBody !== 'object' || Array.isArray(resBody) || !resBody) {
-          return Promise.reject(LOAD_STATUSES.invalidData);
+          // eslint-disable-next-line prefer-promise-reject-errors
+          return Promise.reject({ status: 'invalid_data' });
```

Comment:
It looks like `LOAD_STATUSES.invalidData` is a string:

https://github.com/department-of-veterans-affairs/vets-website/blob/3d6f00d56d7920c13a2e0894c37f51f76594b163/src/platform/forms/save-in-progress/actions.js#L43-L50

Why are we changing the rejection to use an object instead?

Reply:
Good question! I'll walk you through my reasoning:
1. In `fetchInProgressForm`, `apiRequest` now rejects Promises with a `Response` object.
2. Which means that `fetchInProgressForm`'s `catch` now expects a `Response` object, whereas before it expected either a string or Error.
3. Which means that errors thrown within `fetchInProgressForm` must throw an object with a `status` property, as opposed to a `status` string.

I'm not very excited about this solution, and would prefer to re-design `apiRequest` to make the upgrade in this PR easier, but for the reasons outlined in the PR description, I don't think it's a good idea for me to do that now.